### DISCUSSION TITLE
windows/ios client cannot login

### DISCUSSION THREAD

**Discussion by 946106004** [dani-garcia/vaultwarden#3273] [Q&A]
### Deployment environment

* vaultwarden version:
1.27.0

* Install method:
docker run -d --name vaultwarden -v /data/bitwarden/:/data/ -p 443:80  -e ROCKET_TLS='{certs="/data/xxxxx+1.pem",key="/data/xxxxx+1-key.pem"}' vaultwarden/server:latest
The key/pem generate by mkcert command .

* Clients used: 
IOS bitwarden 
* Reverse proxy and version: 

* MySQL/MariaDB or PostgreSQL version: 

* Other relevant details:
Edge plugin can login to server,but IOS/Windows client cannot login,error msg:failed to fetch .And server log:
[2023-02-22 05:37:12.425][rustls::conn][ERROR] TLS alert received: AlertMessagePayload {
    level: Fatal,
    description: CertificateUnknown,
}

![微信图片_20230222135834](https://user-images.githubusercontent.com/52987508/220536079-4b4439a1-04e3-404b-aff8-bdfa870b4f32.png)
![微信图片_20230222140014](https://user-images.githubusercontent.com/52987508/220536084-55f02893-3e1e-4d02-ac91-b8a4d3acac7f.png)

**Comment by stefan0xC** [comment]
You have to [trust the self-signed certificates](https://bitwarden.com/help/certificates/#trust-a-self-signed-certificate). But [as stated in the wiki](https://github.com/dani-garcia/vaultwarden/wiki/Private-CA-and-self-signed-certs-that-work-with-Chrome) using self signed certificates is neither recommended nor a supported method:

> :warning: :hankey: :warning: This method is for testing and development only. The vast majority of users should not use this method, as it requires loading a cert on each of your devices, which is both error-prone and requires future maintenance. Instead, focus your energy on obtaining real certs via [Let's Encrypt](https://letsencrypt.org/getting-started/). This can even work if your vaultwarden instance is not on the public Internet ([example](https://github.com/dani-garcia/vaultwarden/wiki/Running-a-private-vaultwarden-instance-with-Let%27s-Encrypt-certs)).
> :skull_and_crossbones: :skull_and_crossbones: :skull_and_crossbones: This method is not supported. Please do not open GitHub issues or post on the discussion forums asking about how to get this to work.
